hs_composite_resource/models.py

```diff
@@ -117,18 +172,30 @@ def add_file_to_aggregation(self, moved_res_file):
                 # already part of another aggregation (single file aggregation)
                 aggregation.add_resource_file(moved_res_file)
 
-    def get_folder_aggregation_object(self, dir_path):
+    def get_folder_aggregation_object(self, dir_path, aggregations=None):
         """Returns an aggregation (file type) object if the specified folder *dir_path* represents a
          file type aggregation (logical file), otherwise None.
 
          :param dir_path: Resource file directory path (full folder path starting with resource id)
          for which the aggregation object to be retrieved
+         :param aggregations:   list of aggregations in self (resource)
         """
+        if __debug__:
+            if aggregations is not None:
+                assert(isinstance(aggregations, list))
+                for aggr in aggregations:
+                    assert(aggr.resource == self)
+                assert(len(aggregations) == len(list(self.logical_files)))
+
```

Comment:
Statements above for debugging can be removed

Reply:
Fixed